nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/test/java/org/apache/nifi/processors/media/TestExtractMediaMetadata.java

```diff
@@ -0,0 +1,411 @@
+        runner.setProperty(ExtractMediaMetadata.FILENAME_FILTER, ".*[tTxXtT]");
+        runner.setProperty(ExtractMediaMetadata.MIME_TYPE_FILTER, "text/.*");
+        runner.setProperty(ExtractMediaMetadata.METADATA_KEY_FILTER, "");
+        runner.setProperty(ExtractMediaMetadata.METADATA_KEY_PREFIX, "txt.");
+        runner.assertValid();
+
+        runner.run();
+
+        runner.assertAllFlowFilesTransferred(ExtractMediaMetadata.SUCCESS, 0);
+        runner.assertTransferCount(ExtractMediaMetadata.FAILURE, 0);
+    }
+
+    @Test
+    public void testTextFile() throws IOException {
+        final TestRunner runner = TestRunners.newTestRunner(new ExtractMediaMetadata());
+        runner.setProperty(ExtractMediaMetadata.FILENAME_FILTER, ".*[tTxXtT]");
+        runner.setProperty(ExtractMediaMetadata.MIME_TYPE_FILTER, "text/.*");
+        runner.setProperty(ExtractMediaMetadata.METADATA_KEY_FILTER, "");
+        runner.setProperty(ExtractMediaMetadata.METADATA_KEY_PREFIX, "txt.");
+        runner.assertValid();
+
+        runner.enqueue(new File("target/test-classes/notImage.txt").toPath());
+        runner.run();
+
+        runner.assertAllFlowFilesTransferred(ExtractMediaMetadata.SUCCESS, 1);
+        runner.assertTransferCount(ExtractMediaMetadata.FAILURE, 0);
+
+        final List<MockFlowFile> successFiles = runner.getFlowFilesForRelationship(ExtractMediaMetadata.SUCCESS);
+        MockFlowFile flowFile0 = successFiles.get(0);
+        flowFile0.assertAttributeExists("filename");
+        flowFile0.assertAttributeEquals("filename", "notImage.txt");
+        flowFile0.assertAttributeExists("txt.Content-Type");
+        assertTrue(flowFile0.getAttribute("txt.Content-Type").startsWith("text/plain"));
+        flowFile0.assertAttributeExists("txt.X-Parsed-By");
+        assertTrue(flowFile0.getAttribute("txt.X-Parsed-By").contains("org.apache.tika.parser.DefaultParser"));
+        assertTrue(flowFile0.getAttribute("txt.X-Parsed-By").contains("org.apache.tika.parser.txt.TXTParser"));
+        flowFile0.assertAttributeExists("txt.Content-Encoding");
+        flowFile0.assertAttributeEquals("txt.Content-Encoding", "ISO-8859-1");
+        flowFile0.assertContentEquals("This file is not an image and is used for testing the image metadata extractor.".getBytes("UTF-8"));
+    }
+
+    @Test
+    public void testJunkBytes() throws IOException {
+        final TestRunner runner = TestRunners.newTestRunner(new ExtractMediaMetadata());
+        runner.setProperty(ExtractMediaMetadata.FILENAME_FILTER, ".*");
+        runner.setProperty(ExtractMediaMetadata.METADATA_KEY_FILTER, "");
+        runner.setProperty(ExtractMediaMetadata.METADATA_KEY_PREFIX, "junk.");
+        runner.assertValid();
+
+        final Map<String, String> attrs = new HashMap<>();
+        attrs.put("filename", "junk");
+        Random random = new Random();
+        byte[] bytes = new byte[2048];
```

Comment:
Random is not used and the byte[] will always be initialized to be full of zeros[1]. Random should be used to fill it like so: 
        `random.nextBytes(bytes);`

[1] http://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html